src/tls.cpp

```diff
@@ -233,20 +234,21 @@ static inline EVP_PKEY* generate_EC_pkey_with_NID(int nid=NID_secp384r1)
     return result ? pkey : nullptr;
 }
 
-static inline EVP_PKEY* generate_RSA_pkey(void)
+static inline EVP_PKEY* generate_RSA_pkey(unsigned int bits = 2048)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L && !defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
Why you're excluding LibreSSL? AFAIK, is it only a rebranded OpenSSL.

Reply:
@agalanin-at-nvidia LibreSSL has own and incompatible versioning. Check this comment for details: https://github.com/strophe/libstrophe/blob/master/src/tls_openssl.c#L35.

Reply:
Thanks for the explanation! @alexbriskin, probably you need to revert change on this comment.

Reply:
No need to revert. Either keep only openssl version check (libressl reports quite low openssl version and all the features will be disabled with libressl) or make complete version check for libressl version as well. I would prefer the 1st approach, because adding explicit support of libressl is an additional effort and we can do this later if needed.